src/controller/event/handler/internal/artifact.go

```diff
@@ -55,43 +99,137 @@ func (a *Handler) IsStateful() bool {
 }
 
 func (a *Handler) onPull(ctx context.Context, event *event.ArtifactEvent) error {
+	// if duration is equal to 0 or negative, keep original sync mode.
```

Comment:
Does this func always return nil?

Reply:
> Does this func always return nil?

Yes, it always returns nil and only print error logs if meet error.